### ISSUE TITLE
intent to start/stop

### ISSUE THREAD

**Issue by gilcu3** [tfonteyn/Sshd4a#4]
Thank you for the app, I had been waiting for years for something like this to appear.

I personally use the app?to?make?backups from a linux computer using rsync. To automate this it would?be wonderful if your app provided Intents to start and stop the server. Now I am doing it by using the option to start the server by default, but that is less efficient :)

On another topic, what is the option using supersu about? I am not sure the website mentioned in the subtitle is legit, as SuperSU has been deprecated several years already.

**Comment by tfonteyn** [comment]
> intents

I can add those, but I'm curious how you would use them ? Would you not need another app to trigger the intent in the first place? I never looked into this :)

> supersu

to be honest, I only looked at it some 2 years ago when I took my inspiration from the no-longer maintained simplesshd.
I've not tried or looked at it again since.
Do you have more information on it being deprecated?  I briefly searched for more info yesterday, and it's a little confusing. Maybe I should just remove the option in the next release.

**Comment by gilcu3** [comment]
> I can add those, but I'm curious how you would use them ? Would you not need another app to trigger the intent in the first place? I never looked into this :)

I would trigger it for example from `adb`, or from other apps such as `tasker`.

> to be honest, I only looked at it some 2 years ago when I took my inspiration from the no-longer maintained simplesshd. I've not tried or looked at it again since. Do you have more information on it being deprecated? I briefly searched for more info yesterday, and it's a little confusing. Maybe I should just remove the option in the next release.

I think removing it makes sense.

**Comment by tfonteyn** [comment]
> tasker

ah - looks like a good idea.

ok, give me some time, I'm a bit busy this week, but I'll go ahead with your request.

**Comment by tfonteyn** [comment]
sorry for the delay, I've been preoccupied.

I've just released [1.3.0](https://github.com/tfonteyn/Sshd4a/releases/tag/v1.3.0) with support for

"com.hardbacknutter.sshd.fg.START" and  "com.hardbacknutter.sshd.fg.STOP"

This functionality can be enabled/disabled in the settings. Note that when enabled, any other application can send these intents.

**Comment by gilcu3** [comment]
Hi, sorry for the delay, I was just able to test this now with:
```
am start -a com.hardbacknutter.sshd.fg.START
```
which works, but doing:
```
am start -a com.hardbacknutter.sshd.fg.STOP
```
does not. Did you test it with a different command?

**Comment by tfonteyn** [comment]
looks like I [bleep]ed up partially due to my inexperience with external intents.

I did use those commands, and if you test an ssh connection after sending the STOP, you will find the service **has** stopped.
And that is were I left it when testing 👎 

But... the app itself is merely pushed to the background. So if you reselect it from the background, it will use the original intent, i.e. the START command, and restart the service.

I *think* I just found the right solution: replace   Activity.finish()  with  Activity.finishAndRemoveTask();
but I need to check this in detail to prevent another "ouch"....

**Comment by tfonteyn** [comment]
well, this seems to work properly now. Please give [1.3.1](https://github.com/tfonteyn/Sshd4a/releases/download/v1.3.1/Sshd4a-1.3.1.apk) a try.

**Comment by gilcu3** [comment]
@tfonteyn just tested, unfortunately I am seeing exactly the same behavior. Is there any way I can help debug the issue?

**Comment by tfonteyn** [comment]
Can you check these settings **exactly**:

Start the app manually, and goto the "Settings". Make sure the startup configuration is as follows (for this test):
    1. Start... is booted: OFF
    2. Start ... launched: OFF
    3. Allow ... intent:  ON
    4. Foreground...  ON 

Then quit the app, and try the intent commands.

I think the issue will be be that you have option 2 set to 'on' ?
You see, for (some) security, I've designed the 'intent' interaction that the service can ONLY be stopped by the stop-intent **IF** it was started by the start-intent.

If you have for example, set the option 2. "Start when launched" then the order of processing is:
1. Start Intent send by you
2. app starts, and honours the "when launched" option -> service is started.
3. the start-intent is processed... and ignored as the service is already running.
4. You send the stop-intent
5. The stop-intent is ignored because the service was started **at-launch** and NOT by intent.

In other words: do NOT set the service to auto-start, either on-boot, or on-launch if you want to use the Intents.

Hope this makes sense?


PS: 1.3.1 is still needed, due to the need for calling Activity.finish() --> Activity.finishAndRemoveTask();

**Comment by gilcu3** [comment]
Makes sense, tested and works exactly as you mention. Would it make sense to disable those two automatically when the `intent` option is activated? In any case, the explanation above should be enough as well, thank you very much.

**Comment by tfonteyn** [comment]
yeah - I added this to the wiki docs. And you're probably right. Right now there are no checks on the 4 options... something for me to do the next time it rains here :)
